src/Api/Serializer/RestSerializer.php

```diff
@@ -127,6 +127,12 @@ private function applyHeader($name, Shape $member, $value, array &$opts)
             $value = TimestampShape::format($value, 'rfc822');
         }
 
+        if (!empty($member->offsetGet('jsonvalue'))
+            && $member->offsetGet('jsonvalue')
+        ) {
+            $value = base64_encode(json_encode(json_decode($value)));
```

Comment:
Why `decode` then `encode` there? Perhaps you are trying to expect customer provide an valid json string input? If so why not use it directly with `base64_encode`? Also, if we are relying on customers to provide a vaild JSON string, could you somehow raise error if they are providing invalid inputs? Or maybe better documentation for what's expected for those inputs?